### PR TITLE
Improve startup time by speeding up lua module requirements

### PR DESCRIPTION
**Pull request by Pocco81**
### Who is getting benefited?

+ NvChad modules that are required after the impatient plugin (the first plugin to be loaded) gets loaded.
+ Plugins that are not being lazy loaded (e.g. `NvChad/extensions`, `nvim-lua/plenary.nvim`, user plugins they decide not lazy load, etc...)

### Proof:
+ Here is a[ comparison table](https://github.com/lewis6991/impatient.nvim#performance-example) made by the author of the plugin, where it loads plugins with and without impatient.nvim.
+ Personal test: regarding plugins, I don't lazy load `nvim-cmp`: (measured using Packer.nvim's profiler)

Without impatient.nvim:
image

With impatient.nvim:
image

### CODE REVIEW

**Comment by siduck76** [comment]
@Pocco81 I guess we must wait until this stuff gets merged into upstream since we would require lua51-mpack as a dependency for this.

File: lua/plugins/init.lua
Comment (Diego77777777777) [inline lua/plugins/init.lua:5]:
```suggestion
 end
```

Comment (Diego77777777777) [reply lua/plugins/init.lua:5]: Un error

**Review by Diego77777777777 — APPROVED** [review]

**Comment by Diego77777777777** [comment]
****

**Comment by Diego77777777777** [comment]
Te hice varias varios revisiones

**Comment by Diego77777777777** [comment]
> #### This is being done using [impatient.nvim](https://github.com/lewis6991/impatient.nvim).
> 
> ### Who is getting benefited?
> 
> + NvChad modules that are required after the impatient plugin (the first plugin to be loaded) gets loaded.
> + Plugins that are not being lazy loaded (e.g. `NvChad/extensions`, `nvim-lua/plenary.nvim`, user plugins they decide not lazy load, etc...)
> 
> ### Proof:
> + Here is a[ comparison table](https://github.com/lewis6991/impatient.nvim#performance-example) made by the author of the plugin, where it loads plugins with and without impatient.nvim.
> + Personal test: regarding plugins, I don't lazy load `nvim-cmp`: (measured using Packer.nvim's profiler)
> 
> Without impatient.nvim:lynux
>image
> With impatient.nvim:
>image
>

**Comment by Diego77777777777** [comment]
[]()

**Pull request closed without merging**
